src/Test/DepTyCheck/Gen.idr

```diff
@@ -166,20 +134,15 @@ traverseMaybe f (x::xs) = case f x of
   Just y  => (y ::) <$> traverseMaybe f xs
 
 trMTaggedLazy : (a -> Maybe b) -> LazyLst ne (tag, Lazy a) -> Maybe $ LazyLst ne (tag, Lazy b)
-trMTaggedLazy = traverseMaybe . m . wrapLazy where
-  m : (Lazy a -> Lazy (Maybe b)) -> (tag, Lazy a) -> Maybe (tag, (Lazy b))
-  m f (tg, lz) = (tg,) . delay <$> f lz
+trMTaggedLazy = traverseMaybe . wrapMaybeTaggedLazy
 
 trMOneOf : GenAlternatives ne iem a -> (Gen iem a -> Maybe $ Gen em b) -> Maybe $ GenAlternatives ne em b
-trMOneOf (MkGenAlts gs) f = MkGenAlts <$> trMTaggedLazy f gs
+trMOneOf oo f = MkGenAlts <$> trMTaggedLazy f oo.unGenAlts
```

Comment:
Maybe, then, after these simplifications `trMTaggerLazy` should be inlined here, WDYT?

Reply:
I think that makes sense